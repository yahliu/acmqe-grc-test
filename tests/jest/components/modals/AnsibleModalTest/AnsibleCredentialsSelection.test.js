/* Copyright (c) 2020 Red Hat, Inc. */
/* Copyright Contributors to the Open Cluster Management project */

'use strict'

import { renderAnsibleCredentialsSelection } from '../../../../../src-web/components/modals/AnsibleAutomationModal/AnsibleCredentialsSelection'

const data = {
  ansibleCredentials: [
    {
      '__typename': 'AnsibleCredential',
      'name': 'grc-testing-new',
      'namespace': 'default',
      'host': 'https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-jbwbd.dev08.red-chesterfield.com',
      'token': '123456'
    }
  ]
}

describe('render AnsibleCredential component', () => {
  it('render AnsibleCredential component1', () => {
    const component = renderAnsibleCredentialsSelection({
      data, locale: 'us', credentialName: 'grc-testing', credentialIsOpen: false,
      setCredentialsSelectionValue:jest.fn(), onCredentialsSelectionToggle:jest.fn(),
      renderAnsibleJobTemplatesSelection:jest.fn(), getAnsibleConnection:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render AnsibleCredential component2', () => {
    const component = renderAnsibleCredentialsSelection({
      data, locale: 'us', credentialName: 'grc-testing', credentialIsOpen: true,
      setCredentialsSelectionValue:jest.fn(), onCredentialsSelectionToggle:jest.fn(),
      renderAnsibleJobTemplatesSelection:jest.fn(), getAnsibleConnection:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render AnsibleCredential component3', () => {
    const component = renderAnsibleCredentialsSelection({
      data, locale: 'us', credentialName: '', credentialIsOpen: true,
      setCredentialsSelectionValue:jest.fn(), onCredentialsSelectionToggle:jest.fn(),
      renderAnsibleJobTemplatesSelection:jest.fn(), getAnsibleConnection:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })

  it('render empty AnsibleCredential component4', () => {
    const component = renderAnsibleCredentialsSelection({
      data: {ansibleCredentials: ''}, locale: 'us', credentialName: 'grc-testing', credentialIsOpen: false,
      setCredentialsSelectionValue:jest.fn(), onCredentialsSelectionToggle:jest.fn(),
      renderAnsibleJobTemplatesSelection:jest.fn(), getAnsibleConnection:jest.fn()
    })
    expect(component).toMatchSnapshot()
  })
})
