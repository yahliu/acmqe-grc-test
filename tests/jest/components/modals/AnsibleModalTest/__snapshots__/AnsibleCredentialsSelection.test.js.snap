// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render AnsibleCredential component render AnsibleCredential component1 1`] = `
<React.Fragment>
  <React.Fragment>
    <div
      className="titleWithTooltip"
    >
      <Title
        headingLevel="h3"
      >
        Credential
      </Title>
      <div
        className="GrayOutlinedQuestionCircleIcon"
      >
        <mockTooltip
          content={
            <div>
              Credential
            </div>
          }
          position="top"
        >
          <OutlinedQuestionCircleIcon
            color="#6a6e73"
            noVerticalAlign={false}
            size="sm"
          />
        </mockTooltip>
      </div>
    </div>
    <div
      className="infoArea createCredential"
    >
      <Select
        aria-describedby=""
        aria-invalid={false}
        aria-label="Select Ansible credential"
        aria-labelledby=""
        chipGroupComponent={null}
        className=""
        clearSelectionsAriaLabel="Clear all"
        createText="Create"
        customBadgeText={null}
        customContent={null}
        direction="down"
        favorites={Array []}
        favoritesLabel="Favorites"
        hasInlineFilter={false}
        inlineFilterPlaceholderText={null}
        inputIdPrefix=""
        isCreatable={false}
        isDisabled={false}
        isGrouped={false}
        isInputValuePersisted={false}
        isOpen={false}
        isPlain={false}
        menuAppendTo="inline"
        noResultsFoundText="No results found"
        onClear={[Function]}
        onCreateOption={[Function]}
        onFilter={null}
        onSelect={[Function]}
        onToggle={[MockFunction]}
        onTypeaheadInputChanged={null}
        ouiaSafe={true}
        placeholderText="Select Ansible credential"
        removeSelectionAriaLabel="Remove"
        selections="grc-testing"
        toggleAriaLabel="Options menu"
        toggleIcon={null}
        toggleId={null}
        typeAheadAriaLabel=""
        validated="default"
        variant="single"
        width=""
      >
        <SelectOption
          className=""
          component="button"
          description="https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-jbwbd.dev08.red-chesterfield.com"
          index={0}
          inputId=""
          isChecked={false}
          isDisabled={false}
          isFavorite={null}
          isLoad={false}
          isLoading={false}
          isNoResultsOption={false}
          isPlaceholder="grc-testing-new"
          isSelected={false}
          keyHandler={[Function]}
          onClick={[Function]}
          sendRef={[Function]}
          value="grc-testing-new"
        />
      </Select>
      <AcmLaunchLink
        links={
          Array [
            Object {
              "href": "/multicloud/credentials",
              "icon": <AcmIcon
                icon="opennewtab"
              />,
              "id": "createCredentialLink",
              "noIcon": false,
              "text": <TruncateText
                maxCharacters={60}
                text="Add credentials"
              />,
            },
          ]
        }
      />
    </div>
  </React.Fragment>
</React.Fragment>
`;

exports[`render AnsibleCredential component render AnsibleCredential component2 1`] = `
<React.Fragment>
  <React.Fragment>
    <div
      className="titleWithTooltip"
    >
      <Title
        headingLevel="h3"
      >
        Credential
      </Title>
      <div
        className="GrayOutlinedQuestionCircleIcon"
      >
        <mockTooltip
          content={
            <div>
              Credential
            </div>
          }
          position="top"
        >
          <OutlinedQuestionCircleIcon
            color="#6a6e73"
            noVerticalAlign={false}
            size="sm"
          />
        </mockTooltip>
      </div>
    </div>
    <div
      className="infoArea createCredential"
    >
      <Select
        aria-describedby=""
        aria-invalid={false}
        aria-label="Select Ansible credential"
        aria-labelledby=""
        chipGroupComponent={null}
        className=""
        clearSelectionsAriaLabel="Clear all"
        createText="Create"
        customBadgeText={null}
        customContent={null}
        direction="down"
        favorites={Array []}
        favoritesLabel="Favorites"
        hasInlineFilter={false}
        inlineFilterPlaceholderText={null}
        inputIdPrefix=""
        isCreatable={false}
        isDisabled={false}
        isGrouped={false}
        isInputValuePersisted={false}
        isOpen={true}
        isPlain={false}
        menuAppendTo="inline"
        noResultsFoundText="No results found"
        onClear={[Function]}
        onCreateOption={[Function]}
        onFilter={null}
        onSelect={[Function]}
        onToggle={[MockFunction]}
        onTypeaheadInputChanged={null}
        ouiaSafe={true}
        placeholderText="Select Ansible credential"
        removeSelectionAriaLabel="Remove"
        selections="grc-testing"
        toggleAriaLabel="Options menu"
        toggleIcon={null}
        toggleId={null}
        typeAheadAriaLabel=""
        validated="default"
        variant="single"
        width=""
      >
        <SelectOption
          className=""
          component="button"
          description="https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-jbwbd.dev08.red-chesterfield.com"
          index={0}
          inputId=""
          isChecked={false}
          isDisabled={false}
          isFavorite={null}
          isLoad={false}
          isLoading={false}
          isNoResultsOption={false}
          isPlaceholder="grc-testing-new"
          isSelected={false}
          keyHandler={[Function]}
          onClick={[Function]}
          sendRef={[Function]}
          value="grc-testing-new"
        />
      </Select>
      <AcmLaunchLink
        links={
          Array [
            Object {
              "href": "/multicloud/credentials",
              "icon": <AcmIcon
                icon="opennewtab"
              />,
              "id": "createCredentialLink",
              "noIcon": false,
              "text": <TruncateText
                maxCharacters={60}
                text="Add credentials"
              />,
            },
          ]
        }
      />
    </div>
  </React.Fragment>
</React.Fragment>
`;

exports[`render AnsibleCredential component render AnsibleCredential component3 1`] = `
<React.Fragment>
  <React.Fragment>
    <div
      className="titleWithTooltip"
    >
      <Title
        headingLevel="h3"
      >
        Credential
      </Title>
      <div
        className="GrayOutlinedQuestionCircleIcon"
      >
        <mockTooltip
          content={
            <div>
              Credential
            </div>
          }
          position="top"
        >
          <OutlinedQuestionCircleIcon
            color="#6a6e73"
            noVerticalAlign={false}
            size="sm"
          />
        </mockTooltip>
      </div>
    </div>
    <div
      className="infoArea createCredential"
    >
      <Select
        aria-describedby=""
        aria-invalid={false}
        aria-label="Select Ansible credential"
        aria-labelledby=""
        chipGroupComponent={null}
        className=""
        clearSelectionsAriaLabel="Clear all"
        createText="Create"
        customBadgeText={null}
        customContent={null}
        direction="down"
        favorites={Array []}
        favoritesLabel="Favorites"
        hasInlineFilter={false}
        inlineFilterPlaceholderText={null}
        inputIdPrefix=""
        isCreatable={false}
        isDisabled={false}
        isGrouped={false}
        isInputValuePersisted={false}
        isOpen={true}
        isPlain={false}
        menuAppendTo="inline"
        noResultsFoundText="No results found"
        onClear={[Function]}
        onCreateOption={[Function]}
        onFilter={null}
        onSelect={[Function]}
        onToggle={[MockFunction]}
        onTypeaheadInputChanged={null}
        ouiaSafe={true}
        placeholderText="Select Ansible credential"
        removeSelectionAriaLabel="Remove"
        selections=""
        toggleAriaLabel="Options menu"
        toggleIcon={null}
        toggleId={null}
        typeAheadAriaLabel=""
        validated="default"
        variant="single"
        width=""
      >
        <SelectOption
          className=""
          component="button"
          description="https://ansible-tower-web-svc-tower.apps.policy-grc-cp-dev-jbwbd.dev08.red-chesterfield.com"
          index={0}
          inputId=""
          isChecked={false}
          isDisabled={false}
          isFavorite={null}
          isLoad={false}
          isLoading={false}
          isNoResultsOption={false}
          isPlaceholder="grc-testing-new"
          isSelected={false}
          keyHandler={[Function]}
          onClick={[Function]}
          sendRef={[Function]}
          value="grc-testing-new"
        />
      </Select>
      <AcmLaunchLink
        links={
          Array [
            Object {
              "href": "/multicloud/credentials",
              "icon": <AcmIcon
                icon="opennewtab"
              />,
              "id": "createCredentialLink",
              "noIcon": false,
              "text": <TruncateText
                maxCharacters={60}
                text="Add credentials"
              />,
            },
          ]
        }
      />
    </div>
    
  </React.Fragment>
</React.Fragment>
`;

exports[`render AnsibleCredential component render empty AnsibleCredential component4 1`] = `
<React.Fragment>
  <div
    className="infoArea createCredential noExistingCredential"
  >
    <Text>
      Ansible credential is required to create an Ansible job. Create your credential by clicking the following link:
      <AcmLaunchLink
        links={
          Array [
            Object {
              "href": "/multicloud/credentials",
              "icon": <AcmIcon
                icon="opennewtab"
              />,
              "id": "createCredentialLink",
              "noIcon": false,
              "text": <TruncateText
                maxCharacters={60}
                text="Add credentials"
              />,
            },
          ]
        }
      />
    </Text>
  </div>
</React.Fragment>
`;
